examples/gamma_gamma_hier.h

```diff
@@ -0,0 +1,125 @@
+#ifndef BAYESMIX_HIERARCHIES_GAMMAGAMMA_HIERARCHY_H_
+#define BAYESMIX_HIERARCHIES_GAMMAGAMMA_HIERARCHY_H_
+
+#include <google/protobuf/stubs/casts.h>
+#include <src/hierarchies/base_hierarchy.h>
+#include <src/hierarchies/conjugate_hierarchy.h>
+
+#include <Eigen/Dense>
+#include <memory>
+#include <vector>
+
+namespace GammaGamma {
+//! Custom container for State values
+struct State {
+  double rate;
+};
+
+//! Custom container for Hyperparameters values
+struct Hyperparams {
+  double shape, rate_alpha, rate_beta;
+};
+};  // namespace GammaGamma
+
+class GammaGammaHierarchy
+    : public ConjugateHierarchy<GammaGammaHierarchy, GammaGamma::State,
+                                GammaGamma::Hyperparams,
+                                bayesmix::EmptyPrior> {
+ public:
+  GammaGammaHierarchy(double shape, double rate_alpha, double rate_beta)
+      : shape(shape), rate_alpha(rate_alpha), rate_beta(rate_beta) {
+    this->prior = std::make_shared<bayesmix::EmptyPrior>();
+  }
```

Comment:
I don't really think so since this approach does not let you use the factory. I think this example goes more in the direction of showing how easy it is to extend the library for a specific case that we don't implement yet...

Reply:
Fair enough :)